src/index.ts

```diff
@@ -127,7 +180,8 @@ function getEmit(
     // Emit Javascript
     const output = instance.languageService.getEmitOutput(filePath);
 
-    loader.clearDependencies();
+    // Remove original dependency on resourcePath and replace with rawFilePath
```

Comment:
Well my understanding is that the goal of the original call to `clearDependencies` is to replace the dependency on `resourcePath` with `rawFilePath`, because it might be different.

To understand what is going on, I have looked in [webpack/loader-runner](https://github.com/webpack/loader-runner) where dependencies for loaders are handled, and f.ex. `addDependency` and `clearDependencies` are implemented.

It turns out that there is no method that will remove individual dependencies. The only method that actually removes dependencies is `clearDependencies`.

There is a method `getDependencies` that returns file dependencies, but it returns a copy of the internal dependency array, so any changes to will have no effect. The method `getContextDependencies` work in the same way.

So, if I want to replace a dependency, I have to get the existing dependencies, call `clearDependencies`, add all the dependencies except the one I want to replace, and then add the new dependency.

It turns out that `clearDependencies` clears file **_and_** context dependencies, so I have to do the same for context dependencies.